homeassistant/components/lifx/light.py

```diff
@@ -422,16 +426,17 @@ async def register_new_bulb(self, bulb):
                 else:
                     entity = LIFXWhite(bulb, self.effects_conductor)
 
-                _LOGGER.debug("%s register READY", entity.who)
+                _LOGGER.debug("Connected to %s", entity.who)
                 self.entities[bulb.mac_addr] = entity
+                self.discoveries_inflight.pop(bulb.mac_addr, None)
                 self.async_add_entities([entity], True)
```

Comment:
I wonder if the race could be fixed by just not adding the entity if it turns out that it was already added while we were getting the product info. Then we would not need the new dictionary.

```suggestion
                if bulb.mac_addr in self.entities:
                    _LOGGER.debug("Ignoring concurrently connected %s", entity.who)
                else:
                    _LOGGER.debug("Connected to %s", entity.who)
                    self.entities[bulb.mac_addr] = entity
                    self.async_add_entities([entity], True)
```

Reply:
I thought about this but I wanted to stop the second or subsequent attempts before they even start to try and reduce overall network traffic. If we only check here, all that discovery is still being done. 

Let me take another look when I'm not on my phone and back at my desk tomorrow.

Reply:
Yeah, this is too late in the process. The idea behind adding `discoveries_inflight` is tha I don't want Home Assistant to send any additional packets to the bulb.